repair/repair.cc

```diff
@@ -2373,9 +2439,19 @@ future<> repair_service::repair_tablets(repair_uniq_id rid, sstring keyspace_nam
             throw std::runtime_error("The current host must be part of the repair");
         }
 
+        std::ranges::sort(metas, [] (const repair_tablet_meta& a, const repair_tablet_meta& b) {
+            return a.master_shard_id < b.master_shard_id;
+        });
+
         size_t nr = 0;
+        bool metas_added = false;
+        size_t meta_shard_id = 0;
         for (auto& m : metas) {
             nr++;
+            if (m.master_shard_id != meta_shard_id && !shard_task_metas.empty()) {
+                co_await task_metas_builder.add_on_shard(meta_shard_id, std::exchange(shard_task_metas, {}));
```

Comment:
This makes us to accumulate all the metas and add in one go which could be large and introduce the sorting.

I was thinking something  like this. That's why I mentioned batch *a few* metas. 

future<> repair::remote_metas_builder::add_on_shard(size_t shard_id, tablet_repair_task_meta meta) { 

if (_pending_metas[shard_id].size()  < 32) {
    _pending_metas[shard_id].push_back(std::move(meta));
    co_return;
}

Add the _pending_metas[shard_id] with submit_to

}

In repair::remote_metas_builder::build() {
Add any remaining _pending_metas[shard_id]
}

This way we do not any special logic remote_metas_builder, like sorting and call add_on_shard on leave and when a new shard starts.